server/modules/protocol/MySQL/mysql_common.cc

```diff
@@ -1768,3 +1767,190 @@ void mxs_mysql_execute_kill_user(MXS_SESSION* issuer, const char* user, kill_typ
 
     mxs_mysql_send_ok(issuer->client_dcb, info.targets.size(), 0, NULL);
 }
+
+/**
+ *  Parse ok packet to get session track info, save to buff properties
+ *  @param buff           Buffer contain multi compelte packets
+ *  @param packet_offset  Ok packet offset in this buff
+ *  @param packet_len     Ok packet lengh
+ */
+void mxs_mysql_parse_ok_packet(GWBUF *buff,
+                               size_t packet_offset,
+                               size_t packet_len,
+                               uint32_t server_capabilities)
+{
+    uint8_t local_buf[packet_len];
+    uint8_t *ptr = local_buf;
+    char *trx_info, *var_name, *var_value;
+
+    gwbuf_copy_data(buff, packet_offset, packet_len, local_buf);
+    ptr += (MYSQL_HEADER_LEN + 1); // Header and Command type
+    mxs_leint_consume(&ptr);       // Affected rows
+    mxs_leint_consume(&ptr);       // Last insert-id
+    uint16_t server_status = gw_mysql_get_byte2(ptr);
+    ptr += 2; // status      
+    ptr += 2; // number of warnings
+
+    if (server_capabilities & GW_MYSQL_CAPABILITIES_SESSION_TRACK)
```

Comment:
This should be moved before the `while` loop in `mxs_mysql_get_session_track_info` so that if the server doesn't support the capability, the code won't be executed.